contracts/FriendsOnChainV2.sol

```diff
@@ -0,0 +1,167 @@
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+
+// @title Friends on ChainV2
+// @author Ian Hunter (@ianh), Derek Brown (@derekbrown)
+contract FriendsOnChainV2 is Initializable, ERC1155Upgradeable, OwnableUpgradeable {
+  using Counters for Counters.Counter;
+  using Strings for uint256;
+
+  // Defaults to 0. 0 is unlimited supply.
+  uint256 private maxSupply;
+
+  // Indicates the number of people in a FOC, including the minter. Defaults to 7. 0 is unlimited.
+  uint256 private maxOwners;
+
+  // Defaults to free mint. Denominated in wei.
+  uint256 private pricePerToken;
+
+  // Use OpenZeppelin Counters for incrementing - do NOT access counter's underlying value.
+  Counters.Counter private nextTokenId;
+
+  mapping(uint256 => address[]) private ownersOfToken;
+
+  event GroupCreated(uint256 tokenId, address[] indexed _to);
+  event FriendAddedToGroup(uint256 tokenId, address indexed _to);
+
+  /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+  function initialize() initializer public {
+        __ERC1155_init("https://bunches.xyz/foc/metadata/{id}");
+        __Ownable_init();
+        maxSupply = 0;
+        maxOwners = 7;
+        pricePerToken = 0;
+        nextTokenId.increment(); // nextTokenId is initialized to 1, since starting at 0 leads to higher gas cost for the first minter
+
+    }
+
+  /// @notice Mint a token for up to maxOwners addresses.
+  /// @dev Payable. Emits GroupCreated event. Checks maxOwners and maxSupply.
+  /// @dev Loops through addresses to assign ownership.
+  /// @param _to array of recipient addresses of the token
+  function createGroup(address[] memory _to) external payable {
+    uint256 currentTokenId = nextTokenId.current();
+
+    require(
+      maxOwners == 0 || _to.length <= maxOwners,
+      "Maximum number of owners exceeded"
+    );
+    require(msg.value == pricePerToken, "Incorrect payment");
+    require(
+      maxSupply == 0 || currentTokenId <= maxSupply,
+      "Maximum number of tokens reached"
+    );
+
+    for (uint256 i = 0; i < _to.length; i++) {
+      _mint(_to[i], currentTokenId, 1, "");
+    }
+
+    ownersOfToken[currentTokenId] = _to;
+
+    emit GroupCreated(currentTokenId, _to);
+
+    nextTokenId.increment();
+  }
+
+  /// @notice Adds new address to an exisiting Group
+  function addFriendToGroup(address _newFriend, uint256 _tokenId) public payable onlyMemberOrOwner(_tokenId) {
+    require(balanceOf(_newFriend, _tokenId) == 0, "Only 1 of each token is allowed per address");
+    require(msg.value == pricePerToken, "Incorrect payment");
+    address[] memory _ownersOfToken = ownersOfToken[_tokenId]; 
```

Comment:
May want to think about renaming the scoped variable to something like `previousOwners` or `currentOwners` (or something else that's clearer) to avoid accidental confusion between `_ownersOfToken` and `ownersOfToken`.